Assets/Script/Enemy/Enemy.cs

```diff
@@ -10,29 +10,43 @@ public class Enemy : MonoBehaviour
     private Animator _anim;
     private bool _dead = false;
 
-    private string _name = "EnemyDefault";
+    private EnemyHealthBar _enemyHealthBar;
+    private string _name { get; set; }
+
+    protected void Awake()
+    {
+        _name = this.transform.parent.name;
```

Comment:
I am a bit worried that two game object might have the same name. Can backend return a confirmation that the name is unique. Or return a "{name}_{instance_ID}" to the script as its true ID.
![image](https://github.com/SourceGold/SourceGold-Dev/assets/32349680/92bf0b76-6cce-433c-b521-6715c3586b77)

Reply:
This just shows that same name object is allowed

Reply:
Backend will treat any obj with the same name as the same obj since it has no knowledge of whether they are the same or not. Most of the enemies should be generated anyway, so we should make sure we have unique names when generating the names.

Reply:
One thing the backend can do is to throw an exception if the same name is already registered, but that could result in a bad dev experience (if we need to replace an object) if we are willing to work with it.